### PR TITLE
Return error on JSON serialization error.

### PR DESCRIPTION
**Pull request by jeschkies**
Clean up of https://github.com/grafana/loki/pull/7721

### CODE REVIEW

File: pkg/util/marshal/marshal_test.go [hunk 1]
```diff
@@ -722,7 +721,7 @@ func randEntries(rand *rand.Rand) []logproto.Entry {
 func Test_EncodeResult_And_ResultValue_Parity(t *testing.T) {
 	f := func(w wrappedValue) bool {
 		var buf bytes.Buffer
-		js := jsoniter.NewStream(jsoniter.ConfigFastest, &buf, 0)
+		js := json.NewStream(json.ConfigFastest, &buf, 0)
```
Comment (dannykopping) [inline pkg/util/marshal/marshal_test.go:724]: It seems that the test here knows too much about the implementation detail, no?

File: pkg/util/marshal/query.go [hunk 2]
```diff
@@ -264,7 +264,7 @@ func encodeStreams(streams logqlmodel.Streams, s *jsoniter.Stream) error {
 
 		err := encodeStream(stream, s)
 		if err != nil {
-			return nil
+			return err
```
Comment (dannykopping) [inline pkg/util/marshal/query.go:267]: Did a test fail in CI and that's why we're adding this? Or do we need another to test this?

Comment (chaudum) [reply pkg/util/marshal/query.go:267]: Yes, ideally we have a test case for the error case as well.

Comment (jeschkies) [reply pkg/util/marshal/query.go:267]: Alas. You are right...

Comment (jeschkies) [reply pkg/util/marshal/query.go:274]: ...and there was a bug indeed.

File: pkg/util/marshal/query.go [hunk 3]
```diff
@@ -185,28 +185,35 @@ func EncodeResult(v logqlmodel.Result, s *jsoniter.Stream) error {
...
 
 	s.WriteObjectField("resultType")
 	s.WriteString(string(v.Data.Type()))
 
 	s.WriteMore()
 	s.WriteObjectField("result")
-	encodeResult(v.Data, s)
+	err := encodeResult(v.Data, s)
+	if err != nil {
+		return err
```
Comment (dannykopping) [inline pkg/util/marshal/query.go:207]: Can we wrap these errors for more clarity?

**Review by dannykopping — APPROVED** [review]
LGTM, with a clarity nit

**Pull request merged**